pkg/minikube/cluster/cluster.go

```diff
@@ -97,8 +99,10 @@ func StartHost(api libmachine.API, config MachineConfig) (*host.Host, error) {
 		return nil, fmt.Errorf("Error configuring authorization on host: %s", err)
 	}
 
-	if err := registration.RegisterHostVM(h, RegistrationParameters); err != nil {
-		return nil, fmt.Errorf("Error registering the VM: %s", err)
+	if !minishiftConfig.InstanceConfig.Register {
```

Comment:
It is really unfortunate that this code reaches out to other parts of the code and relies on global variables being set and all. This breaks encapsulation. Really, the method should get passed all the information it needs and should be independent of minishiftConfig.InstanceConfig.

Can we maybe refactor this?